### PR TITLE
ci: add support for develop branch

### PR DESCRIPTION
**Pull request by HarshCasper**

### CODE REVIEW

File: .github/workflows/kubernetes_test.yaml [hunk 1]
```diff
@@ -32,9 +33,9 @@ jobs:
       run:
         shell: bash -l {0}
     steps:
-      - name: "Set QHUB_IMAGE_TAG=main"
+      - name: "Set QHUB_IMAGE_TAG=develop"
```
Comment (trallard) [inline .github/workflows/kubernetes_test.yaml:36]: This hardcodes for `develop` and drops `main` altogether - the idea is that it works for both not one or the other. 
Could leverage GH actions context to get the branch/ref name and set here

File: .github/workflows/kubernetes_test.yaml [hunk 2]
```diff
@@ -32,9 +33,9 @@ jobs:
...
         run: |
-          echo "QHUB_IMAGE_TAG=main" >> "$GITHUB_ENV"
+          echo "QHUB_IMAGE_TAG=develop" >> "$GITHUB_ENV"
```
Comment (trallard) [inline .github/workflows/kubernetes_test.yaml:38]: same as above

**Review by trallard — CHANGES_REQUESTED** [review]

**Comment by trallard** [comment]
After going over the release workflow - and migration tasks - it does seem we need the workflow to run against `main` AND release branches, so we'd need something like:


`branches: [ main, 'release/v[0-9].[0-9]+' ]` cc @costrouc  ofc need to agree on the `release` prefix - but I think that should be a reasonable branching strategy WDYT?

File: .github/workflows/run-pre-commit.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ name: Run pre-commit
 
 on:
   push:
-    branches: [ main ]
+    branches: [ main, develop ]
```
Comment (trallard) [inline .github/workflows/run-pre-commit.yaml:5]: we no longer have a develop branch - there is only a release (with the corresponding CalVer patter) and a main branch

**Comment by HarshCasper** [comment]
Hi @trallard — Can you please have a look at it once again?

**Pull request closed without merging**